File: code/session-1/c++/interactive-example.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
  int high,low;
  string state;
  low = 0;
  cin >> high;
  do {
    int mid;
    mid = (low+high) / 2;
    cout << mid << endl << flush;
    cin >> state;
    if(state.compare("higher") == 0) {
      low = mid +1;
    } else if(state.compare("lower") ==0 ){
      high = mid - 1;
    }
  }while (state.compare("correct") != 0);
  return 0;
}

File: code/session-1/kotlin/interactive-example.kt
fun main() {
  val output = System.out.bufferedWriter()
  var low = 0
  var high = readln().toInt()
  var state: String
  do {
    val mid = (low + high) shr 1
    output.write("$mid\n")
    output.flush()
    state = readln()
    when (state) {
      "higher" -> low = mid + 1
      "lower" -> high = mid - 1
    }
  } while (state != "correct")
}
